test/unit/FormulateInputButton.test.js

```diff
@@ -78,3 +78,13 @@ test('type "button" renders slot inside button', () => {
   })
   expect(wrapper.find('button > span').html()).toBe('<span>My custom slot</span>')
 })
+
+it('passes an explicitly given name prop through to the root element', () => {
+  const wrapper = mount(FormulateInput, { propsData: { type: 'button', name: 'foo' } })
+  expect(wrapper.find('button[name="foo"]').exists()).toBe(true)
+})
+
+it('additional context does not bleed through to button input attributes', () => {
+    const wrapper = mount(FormulateInput, { propsData: { type: 'button' } } )
+    expect(Object.keys(wrapper.find('button').attributes())).toEqual(["type", "id"])
```

Comment:
```suggestion
    expect(Object.keys(wrapper.find('button').attributes()).includes('name')).toBeFalsy()
```
Suggesting we only check for the absence of name so if/when we add other default attributes these tests dont fail.

Reply:
☝️ this tweak could be extended to the other tests where we are check for absence.

Reply:
Do we only want to check for the absence of name? Are we sure we don't want to explicitly whitelist props that should come through in default cases? Changing the logic to only check for the absence of name would mean that if something else leaked through we would not know. Personally I'd be ok with updating the tests to match new default cases when needed.